src/interceptors/fetch/index.ts

```diff
@@ -103,7 +104,10 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
         this.logger.info('received mocked response:', mockedResponse)
 
         // Reject the request Promise on mocked "Response.error" responses.
-        if (mockedResponse.type === 'error') {
+        if (
+          isPropertyAccessible(mockedResponse, 'type') &&
```

Comment:
Can you please add a comment block that explains why we check this so, and include a link to the issue/PR? Thanks.